packages/next/build/webpack/plugins/build-manifest-plugin.ts

```diff
@@ -12,19 +12,43 @@ import {
 import { BuildManifest } from '../../../next-server/server/get-page-files'
 import getRouteFromEntrypoint from '../../../next-server/server/get-route-from-entrypoint'
 import { ampFirstEntryNamesMap } from './next-drop-client-page-plugin'
+import { Rewrite } from '../../../lib/load-custom-routes'
+import {
+  isDynamicRoute,
+  getSortedRoutes,
+} from '../../../next-server/lib/router/utils'
 
 const isWebpack5 = parseInt(webpack.version!) === 5
 
 // This function takes the asset map generated in BuildManifestPlugin and creates a
 // reduced version to send to the client.
 function generateClientManifest(
   assetMap: BuildManifest,
-  isModern: boolean
+  isModern: boolean,
+  rewrites: Rewrite[]
 ): string {
-  const clientManifest: { [s: string]: string[] } = {}
+  const clientManifest: { [s: string]: string[] | Rewrite[] } = {
```

Comment:
IMO this needs to be an array and not an object. It's not intuitive that we rely on ordering of the object keys.

Reply:
Makes sense, I updated to provide the sorted page keys as an additional field instead of changing the structure of the manifest since having to loop over the items for each page asset look-up seems not ideal